File: Examen/app/src/main/java/com/example/examen/MainActivity.java
package com.example.examen;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {

    private final static String CONTADOR_STRING = "Contador ";

    int contador=0;
    EditText et;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        et = ((EditText) findViewById(R.id.editText));
    }

    public void contar(View view) {
        contador++;
        et.setText(CONTADOR_STRING+contador);
    }

    public void enviar(View view) {
        Intent intent = new Intent(this, SecondActivity.class);
        intent.putExtra("text",et.getText().toString());
        startActivity(intent);
    }

    public void toRegistroActivity(View view) {
        startActivity(new Intent(this, RegistroActivity.class));
    }
}

File: MyApplication/app/src/main/java/com/example/myapplication/MainActivity.kt
package com.example.myapplication

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val persona = Programador().getProgrammerData()
        Log.d(":::tag","Mi nombre es ${persona.name}")
        Log.d(":::tag","Tengo ${persona.age} años")
        Log.d(":::tag","Estoy aprendiendo ${persona.language}")
    }

    class Programador(): ProgramadorInterface{
       private fun nombre():String{
           val name: String = "Estefanía"
            return name
        }

        private fun edad():Int{
            val age: Int = 31
            return age
        }

        private fun lenguaje():String{
            val lenguage: String = "Kotklin"
            return lenguage
        }


     override fun getProgrammerData():ProgrammerData{
           val solucion = ProgrammerData(nombre(),edad(),lenguaje())
         return solucion
     }

    }

    interface ProgramadorInterface{
       fun getProgrammerData():ProgrammerData
    }

    data class ProgrammerData(val name: String, val age:Int, val language:String){
    }
}